Snowflake.Data/Core/SFResultSetMetaData.cs

```diff
@@ -37,24 +37,20 @@ class SFResultSetMetaData
         /// </summary>
         private Dictionary<string, int> columnNameToIndexCache = new Dictionary<string, int>();
 
-        internal SFResultSetMetaData(QueryExecResponseData queryExecResponseData)
+        internal SFResultSetMetaData(QueryExecResponseData queryExecResponseData, SFSession session)
         {
             rowTypes = queryExecResponseData.rowType;
             columnCount = rowTypes.Count;
             statementType = findStatementTypeById(queryExecResponseData.statementTypeId);
             columnTypes = InitColumnTypes();
-            
-            foreach (NameValueParameter parameter in queryExecResponseData.parameters)
+
+            if (session.ParameterMap.ContainsKey(SFSessionParameter.DATE_OUTPUT_FORMAT))

```

Comment:
If we have only these two params involved, maybe we should have tests for both cases?

Reply:
Currently TIME_OUTPUT_FORMAT is not used in the driver.
https://github.com/snowflakedb/snowflake-connector-net/blob/v2.1.2/Snowflake.Data/Core/SFBaseResultSet.cs#L55
Not sure whether that's intentional but I think if that needs to be fixed it should be made in a separated PR. In my opinion better not to change it until we get customer issue about that. Time (or maybe timestamp as well) is pretty common data type so likely the existing customer applications already adapted with the existing output format.